packages/common-all/src/dnode.ts

```diff
@@ -807,6 +807,21 @@ export class NoteUtils {
     return out;
   }
 
+  static getNotesByFnameV2({
+    fname,
+    engine,
+    vault,
+  }: {
+    fname: string;
+    engine: DEngineClient;
+    vault?: DVault;
+  }): NoteProps[] {
+    let notes = engine.noteFnames.get(engine.notes, fname);
+    if (vault)
+      notes = notes.filter((note) => VaultUtils.isEqualV2(note.vault, vault));
+    return notes;
+  }
+
   static getNoteByFnameV5({
```

Comment:
If we do these kinds of revision bumps, we should mark the old versions with an `@deprecated` tag. Otherwise, new code will likely continue using v5 as we scan the code base to see what other code is doing as an example(and seeing usages of v5).